src/main/java/org/jenkinsci/plugins/maven_artifact_choicelistprovider/AbstractMavenArtifactChoiceListProvider.java

```diff
@@ -91,9 +132,10 @@ public List<String> getChoiceList() {
      *            the internal jenkins id for the credentials
      * @return the credentials for the ID or NULL
      */
-    public static UsernamePasswordCredentialsImpl getCredentials(final String pCredentialId) {
-        return CredentialsMatchers.firstOrNull(
-                CredentialsProvider.lookupCredentials(UsernamePasswordCredentialsImpl.class, Jenkins.getInstance(), ACL.SYSTEM, Collections.<DomainRequirement> emptyList()),
+    public static UsernamePasswordCredentialsImpl getCredentials(@Nonnull String pCredentialId, @Nonnull Item pItem) {
+		final Authentication acl = pItem instanceof Queue.Task ? Tasks.getAuthenticationOf((Queue.Task) pItem) : ACL.SYSTEM;
+		return CredentialsMatchers.firstOrNull(
+                CredentialsProvider.lookupCredentials(UsernamePasswordCredentialsImpl.class, Jenkins.get(), acl, Collections.<DomainRequirement> emptyList()),
```

Comment:
It's not the Authentication that's passed that is a problem, but the context.
```suggestion
                CredentialsProvider.lookupCredentials(UsernamePasswordCredentialsImpl.class, pItem, ACL.SYSTEM, Collections.<DomainRequirement> emptyList()),
```